ktor-client/ktor-client-core/jvm/src/io/ktor/client/HttpClientJvm.kt

```diff
@@ -34,14 +34,18 @@ public interface HttpClientEngineContainer {
     public val factory: HttpClientEngineFactory<*>
 }
 
-/**
- * Workaround for dummy android [ClassLoader].
- */
-private val engines: List<HttpClientEngineContainer> = HttpClientEngineContainer::class.java.let {
-    ServiceLoader.load(it, it.classLoader).toList()
-}
+// ServiceLoader should use specific call convention to be optimized by R8 on Android:
```

Comment:
Hey @osipxd!
I saw you created an issue (https://youtrack.jetbrains.com/issue/KTOR-7698) and wanted to tell, that I remembered doing it :)

and so... here are the changes. I've totally forgotten to extract them into a separate PR :(
I will then proceed with extracting those changes in separate PR.
Note: the pattern used here is implemented based on https://github.com/whyoleg/sweet-spi/pull/2 where I've already performed some extensive testing, that it works :)
Just telling it, so that you will not need to do this one more time :)